src/collection/domain/repositories/CollectionRepository.ts

```diff
@@ -0,0 +1,5 @@
+import { Collection } from '../models/Collection'
+
+export interface CollectionRepository {
+  getById: (id: string) => Promise<Collection | undefined>
```

Comment:
Returning Collection | undefined in TypeScript works like Java's Optional. It means the 'get' function could return a Collection if one is found, or undefined if not. This ensures type safety: just as you must unwrap an Optional in Java to access its value, in TypeScript, checking for undefined ensures you're aware when a value might not be present. If the Collection isn't found, we typically show a 'Page Not Found' message. For other errors, we throw an exception

Using this type definition, when we create a hook to get the collection, we acknowledge that the collection might not be found and could be undefined. Therefore, we need to handle the possibility of undefined when accessing the collection's properties, ensuring our code safely deals with the absence of a value.

It's a matter of design choice, but another approach could be to always return a Collection and throw an error for any scenario where this isn't possible

Reply:
I think js-dataverse should return `undefined` if it can't find a dataset or any other entity. This way, users know when something doesn't exist and can decide what to do next, like showing an error page or something else. It adds an extra step, but I believe it's useful. However, I might see it this way because of how we do things on the frontend.

We should talk about this with @GPortas. If we decide against returning `undefined` in js-dataverse, we'll just handle that part on the frontend.

Reply:
@ekraffmiller @MellyGray 

I think js-dataverse should always throw errors for any API operation that returns an error. Since it's an API wrapper, IMO it seems more natural to return an error to the consumer than an undefined value, like the API does (the API doesn't return a 200 response with a null body if a resource is not found). Note that the message accompanying the 404 error returned by the API may provide useful information to the consumer, which would be lost if we return undefined.

Regarding what Melina says about ensuring our code safely handles the absence of a value, this can still be done by throwing an error. In js-dataverse there are different types of errors defined, very high level and general purpose for now: https://github.com/IQSS/dataverse-client-javascript/tree/develop/src/core/domain/repositories. For now any error returned by the API in a read operation returns a ReadError, which includes the message with the HTTP error code as a very simple string, but we can define a new NotFoundError that extends ReadError and extend these errors to contain more refined data. This way we can safely identify a not found error within the block catch of the promise, and carry out the necessary actions in the SPA.

Reply:
@GPortas sounds good! How do we proceed with this PR? Should I remove the undefined from the repository definition or keep it until we implement the NotFoundError in js-dataverse?

Reply:
@MellyGray I would remove the undefined type in this PR and then create two new issues:

1) Create the NotFoundError in js-dataverse (js-dataverse repository)
2) Handle the NotFoundError properly in the frontend (dependent on the previous one) (SPA repository)

Please tell me what you think!

Reply:
@MellyGray @ekraffmiller I just created the issues:

- https://github.com/IQSS/dataverse-client-javascript/issues/142
- https://github.com/IQSS/dataverse-frontend/issues/363

Reply:
I removed the undefined